File: app/src/main/java/com/vo1d/schedulemanager/v2/ui/subjects/list/SubjectsListViewModel.java
package com.vo1d.schedulemanager.v2.ui.subjects.list;

import android.app.Application;

import androidx.annotation.NonNull;

import com.vo1d.schedulemanager.v2.data.subjects.Subject;
import com.vo1d.schedulemanager.v2.ui.ListFragmentViewModel;

public class SubjectsListViewModel extends ListFragmentViewModel<Subject> {

    public SubjectsListViewModel(@NonNull Application application) {
        super(application);
    }
}

File: app/src/main/kotlin/ru/vo1dmain/timetables/sections/subjects/list/SubjectsListViewModel.kt
package ru.vo1dmain.timetables.sections.subjects.list

import android.app.Application
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.launch
import ru.vo1dmain.timetables.data.entities.subject.SubjectsRepository
import ru.vo1dmain.timetables.ui.selection.SelectableListViewModel

internal class SubjectsListViewModel(application: Application) :
    SelectableListViewModel<Long>(application) {
    private val repo = SubjectsRepository(application)
    
    val all by lazy { repo.all }
    
    fun deleteAll() {
        viewModelScope.launch {
            repo.deleteAll()
        }
    }
}